src/panels/lovelace/common/validate-condition.ts

```diff
@@ -109,8 +235,49 @@ function checkOrCondition(condition: OrCondition, hass: HomeAssistant) {
   return condition.conditions.some((c) => checkConditionsMet([c], hass));
 }
 
+/**
+ * Build a condition for filters
+ * @param condition condition to apply
+ * @param entityId base the condition on that entity (current entity to filter)
+ * @returns a new condition that handled legacy filter conditions
+ */
+export function buildConditionForFilter(
+  condition: Condition | LegacyFilterCondition | string | number,
+  entityId: string
+): Condition | LegacyFilterCondition {
+  let newCondition: Condition | LegacyFilterCondition;
+
+  if (typeof condition === "string" || typeof condition === "number") {
+    newCondition = {
+      condition: "state",
+      state: condition,
+    } as StateCondition;
+  } else {
+    newCondition = condition;
+  }
+
+  // Set the entity to filter on
+  if (
+    "condition" in newCondition &&
+    (newCondition.condition === "numeric_state" ||
+      newCondition.condition === "state")
+  ) {
+    newCondition.entity = entityId;
+  } else if ("operator" in newCondition) {
+    newCondition.entity = entityId;
+  }
```

Comment:
```suggestion
  if (
    "condition" in newCondition &&
    (newCondition.condition === "numeric_state" ||
      newCondition.condition === "state") ||
    "operator" in newCondition
  ) {
    newCondition.entity = entityId;
  }
```

Reply:
We can merge the conditions

Reply:
Suggestion applied on 69310d2d4593455b9e5771c9c7c336095ed78a9d